src/utils/deferred.ts

```diff
@@ -31,6 +31,9 @@ export class Deferred<T> implements Promise<T> {
       this.#resolve = resolve;
       this.#reject = reject;
     });
+    // Needed to avoid `Uncaught (in promise)`. The promises returned by `then`
```

Comment:
Does it make a sense to add an unit test for this?

Reply:
good point. Done.